src/answers-headless.ts

```diff
@@ -320,7 +320,19 @@ export default class AnswersHeadless {
       context,
       referrerPageUrl,
       restrictVerticals
-    });
+    };
+
+    let response: UniversalSearchResponse;
+    try {
+      response = await this.core.universalSearch(request);
+    } catch (e) {
+      const latestResponseId = this.httpManager.getLatestResponseId('universalQuery');
+      if (thisRequestId > latestResponseId) {
+        this.stateManager.dispatchEvent('searchStatus/setIsLoading', false);
+        this.httpManager.setResponseId('universalQuery', thisRequestId);
+      }
+      return Promise.reject(e);
```

Comment:
just out of curiosity does `throw e` here have the same behavior as returning a Promise.reject(e)?

Reply:
Also what information does this error have? like do we need to add any extra info onto it?

Reply:
I believe so, both would stop the remaining code from running, and requires some rejected promise handling logic from the caller (it will receive a rejected promise in both case)

Reply:
I think the error message coming from core should be sufficient, unless you have something else in mind we should add?

Reply:
Ah I just wasn't sure what the message was. I was mostly wondering whether it said what kind of request failed. The API doesn't always have the best error messages from what I remember

Reply:
Though if the error msg doesn't have that info maybe that's core's job

Reply:
just tested with a failed response from filterSearch, although you can look at the stack trace and network, it looks like the error msg itself doesn't know what kind of request it is. I think this is something we can update in core instead tho

Reply:
I'll make an item!